packages/buidler-core/src/internal/core/providers/gas-providers.ts

```diff
@@ -115,6 +115,25 @@ export function createGanacheGasMultiplierProvider(
   });
 }
 
+let cachedGasLimit: number | undefined;
+async function getGasLimit(provider: IEthereumProvider): Promise<number> {
```

Comment:
`getBlockGasLimit` maybe? `gasLimit` is also used for transactions, so it's confusing.